### PR TITLE
Fix/next 244 - Asset Details API - refactoring the clients to improve modularity and dependency management

### PR DESCRIPTION
**Pull request by plyubich**
This pull request includes significant changes to the `ui-asset-management/api/svc-asset-details/extension/clients` package. The most important changes involve refactoring the clients to improve modularity and dependency management, as well as adding new functionality for handling AWS credentials and configurations.

### Refactoring and Dependency Management:

* [`ui-asset-management/api/svc-asset-details/extension/clients/asset_details_client.go`](diffhunk://#diff-fadf0916add87dd05a7cd3a965cb1339211439a3af02f858ecd50250a72e3537R11-R66): Refactored `NewAssetDetailsClient` to initialize `DynamodbClient` and `SecretsClient` with configuration details, and removed logging dependencies.
* [`ui-asset-management/api/svc-asset-details/extension/clients/elasticsearch_client.go`](diffhunk://#diff-4a3d93386d8804424496b803046e6c90aaa707360c5244e81bb7e7aa6b928674L8-R44): Introduced `NewElasticSearchClient` to initialize with `DynamodbClient`, and added caching for Elasticsearch clients. [[1]](diffhunk://#diff-4a3d93386d8804424496b803046e6c90aaa707360c5244e81bb7e7aa6b928674L8-R44) [[2]](diffhunk://#diff-4a3d93386d8804424496b803046e6c90aaa707360c5244e81bb7e7aa6b928674L31-R58)
* [`ui-asset-management/api/svc-asset-details/extension/clients/rds_client.go`](diffhunk://#diff-fdb2b06a9a69669321c51190b71f452a65d1bea52a85d7807fed132056de354dL30-R55): Refactored `NewRdsClient` to use `SecretsClient` for fetching RDS credentials and added client caching. [[1]](diffhunk://#diff-fdb2b06a9a69669321c51190b71f452a65d1bea52a85d7807fed132056de354dL30-R55) [[2]](diffhunk://#diff-fdb2b06a9a69669321c51190b71f452a65d1bea52a85d7807fed132056de354dL50)

### AWS Credentials and Configuration Handling:

* [`ui-asset-management/api/svc-asset-details/extension/clients/dynamodb_client.go`](diffhunk://#diff-92e0ea5633de71cc7653fbda7b4a859e0b8bbf39be6c3a6bebe85837a0c7153cR22-R111): Updated `NewDynamoDBClient` to use AWS STS for assuming roles and fetching credentials, and refactored `GetOpenSearchDomain` to use the new AWS SDK v2.
* [`ui-asset-management/api/svc-asset-details/extension/clients/config_client.go`](diffhunk://#diff-4487a0463fe4506a1e4da78909e835cd7c8a5da3528e06e125b62d0765928e00L20-R59): Added new configuration fields for `AssumeRoleArn` and `SecretIdPrefix`, and refactored `LoadConfigurationDetails` to load these from environment variables.

### Testing:

* [`ui-asset-management/api/svc-asset-details/extension/clients/dynamodb_client_test.go`](diffhunk://#diff-524970e7c0fbb1a7bd41756a751dfc254e552d89b6542d5b18794153d06c3ebbR1-R45): Added a new test for `NewDynamoDBClient` to verify the initialization and fetching of OpenSearch domain properties.
* [`ui-asset-management/api/svc-asset-details/extension/clients/rds_client_test.go`](diffhunk://#diff-97cb555280d3f971d69a4a80d57e6e631b4f879ec842639df763a64c3e085520R1-R47): Added a new test for `NewRdsClient` to verify fetching mandatory tags from RDS.

These changes aim to improve the modularity, maintainability, and security of the asset management service by better handling dependencies and credentials.

## Release Notes

- **New Features**
  - Comprehensive rewrite of the `README.md` for improved clarity and organization.
  - Added a detailed guide for local setup regarding Paladin Cloud Integration.
  - Introduced caching mechanisms for database and Elasticsearch clients to enhance performance.
  - New data structures for managing tenant-related data and OpenSearch domain properties.
  - Enhanced server functionality with conditional execution based on new parameters.

- **Bug Fixes**
  - Improved error handling across various client implementations.

- **Documentation**
  - Enhanced documentation to provide clearer instructions and examples for users.

- **Tests**
  - Added new test files and functions to validate the functionality of newly implemented clients and methods.

- **Chores**
  - Updated dependencies to transition from AWS SDK v1 to v2 for improved performance and functionality.

### CODE REVIEW

File: ui-asset-management/api/svc-asset-details/extension/clients/secrets_client.go [hunk 1]
```diff
@@ -20,53 +20,86 @@ import (
...
+// GetRdsSecret retrieves the RDS secret from AWS Secrets Manager
+func (r *SecretsClient) GetRdsSecret(ctx context.Context, secretIdPrefix, tenantId string) (*models.RdsSecret, error) {
+	// Create the secretId using the prefix and tenantId
+	secretId := fmt.Sprintf("{%s}{%s}", secretIdPrefix, tenantId)
+	fmt.Printf("Getting Rds Secrets from {%s}\n", secretId)
+
+	// Prepare the input for retrieving the secret
 	input := &secretsmanager.GetSecretValueInput{
-		SecretId:     aws.String(secretName),
+		SecretId:     aws.String("paladincloud/secret/98c28482-9bae-46bd-bd4e-58fa132e72c0"),
```
Comment (kevin-paladin) [inline ui-asset-management/api/svc-asset-details/extension/clients/secrets_client.go:80]: Did you mean to replace the hard-coded string with `secretId` that was created a few lines above?

**Review by kevin-paladin — CHANGES_REQUESTED** [review]
There's a minor change I'd like to see; it all looks good

**Review by kevin-paladin — COMMENTED** [review]
Looks good!

File: ui-asset-management/api/svc-asset-details/extension/main_test.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		config := &clients.Configuration{
+			Region:                        "us-east-1",
+			TenantConfigTable:             "Tenant-Config",
+			TenantConfigTablePartitionKey: "tenantId",
+			SecretIdPrefix:                "paladincloud/secret/",
+		}
+
+		startMain(config)
+	})
+}
```
[the remark this answers is not in the document]
Comment (plyubich) [reply ui-asset-management/api/svc-asset-details/extension/main_test.go]: yes, implement this test

**Review by kevin-paladin — APPROVED** [review]

**Pull request merged**